pkg/blobplugin/main.go

```diff
@@ -81,31 +81,34 @@ func handle() {
 	driverOptions := blob.DriverOptions{
 		NodeID:                                 *nodeID,
 		DriverName:                             *driverName,
-		CloudConfigSecretName:                  *cloudConfigSecretName,
-		CloudConfigSecretNamespace:             *cloudConfigSecretNamespace,
 		BlobfuseProxyEndpoint:                  *blobfuseProxyEndpoint,
 		EnableBlobfuseProxy:                    *enableBlobfuseProxy,
 		BlobfuseProxyConnTimout:                *blobfuseProxyConnTimout,
 		EnableBlobMockMount:                    *enableBlobMockMount,
-		CustomUserAgent:                        *customUserAgent,
-		UserAgentSuffix:                        *userAgentSuffix,
-		AllowEmptyCloudConfig:                  *allowEmptyCloudConfig,
 		EnableGetVolumeStats:                   *enableGetVolumeStats,
 		AppendTimeStampInCacheDir:              *appendTimeStampInCacheDir,
 		MountPermissions:                       *mountPermissions,
 		AllowInlineVolumeKeyAccessWithIdentity: *allowInlineVolumeKeyAccessWithIdentity,
 		AppendMountErrorHelpLink:               *appendMountErrorHelpLink,
-		KubeAPIQPS:                             *kubeAPIQPS,
-		KubeAPIBurst:                           *kubeAPIBurst,
 		EnableAznfsMount:                       *enableAznfsMount,
 		VolStatsCacheExpireInMinutes:           *volStatsCacheExpireInMinutes,
 		SasTokenExpirationMinutes:              *sasTokenExpirationMinutes,
 	}
-	driver := blob.NewDriver(&driverOptions)
+
+	userAgent := blob.GetUserAgent(driverOptions.DriverName, *customUserAgent, *userAgentSuffix)
+	klog.V(2).Infof("driver userAgent: %s", userAgent)
+
+	cloud, err := blob.GetCloudProvider(*kubeconfig, driverOptions.NodeID, *cloudConfigSecretName, *cloudConfigSecretNamespace, userAgent, *allowEmptyCloudConfig, *kubeAPIQPS, *kubeAPIBurst)
```

Comment:
why not put all these into `DriverOptions`? that's the reason why `DriverOptions` exists, it could save parameter num
```
*cloudConfigSecretName, *cloudConfigSecretNamespace, userAgent, *allowEmptyCloudConfig, *kubeAPIQPS, *kubeAPIBurst
```

Reply:
It's a little bit complicated to add mock for GetCloudProvider

Reply:
but why move part of the options as parameters? this is not consistent

Reply:
Should I move other dependencies out of newDriver function? In that case that would be easier to compose driver object and there is no need to implement FakeDriver.